transmogrifier/sources/xml/datacite.py

```diff
@@ -339,6 +238,189 @@ def _get_contributors(cls, source_record: Tag) -> Iterator[timdex.Contributor]:
+
+    @classmethod
+    def _parse_date_range(
+        cls, date_object: timdex.Date, date_value: str, source_record_id: str
+    ) -> timdex.Date:
+        split = date_value.index("/")
+        gte_date = date_value[:split].strip()
+        lte_date = date_value[split + 1 :].strip()
+        if validate_date_range(
+            gte_date,
+            lte_date,
+            source_record_id,
+        ):
+            date_object.range = timdex.DateRange(
+                gte=gte_date,
+                lte=lte_date,
+            )
+        return date_object
+
+    @classmethod
+    def get_edition(cls, source_record: Tag) -> str | None:
+        if edition := source_record.metadata.find("version", string=True):
+            return str(edition.string)
+        return None
+
+    @classmethod
+    def get_file_formats(cls, source_record: Tag) -> list[str] | None:
+        return [
+            str(file_format.string)
+            for file_format in source_record.metadata.find_all("format", string=True)
+        ] or None
+
+    @classmethod
+    def get_format(cls) -> str:
+        return "electronic resource"
+
+    @classmethod
+    def get_funding_information(cls, source_record: Tag) -> list[timdex.Funder] | None:
+        funding_information = []
+        for funding_reference in source_record.metadata.find_all("fundingReference"):
+            funder = timdex.Funder()
+            if funder_name := funding_reference.find("funderName", string=True):
+                funder.funder_name = str(funder_name.string)
+            if award_number := funding_reference.find("awardNumber"):
+                funder.award_number = award_number.string or None
+                funder.award_uri = award_number.get("awardURI") or None
+            if funder_identifier := funding_reference.find(
+                "funderIdentifier", string=True
+            ):
+                funder.funder_identifier = str(funder_identifier.string)
+                funder.funder_identifier_type = (
+                    funder_identifier.get("funderIdentifierType") or None
+                )
+            if funder != timdex.Funder():
+                funding_information.append(funder)
+        return funding_information or None
+
+    @classmethod
+    def get_identifiers(
+        cls,
+        source_record: Tag,
+    ) -> list[timdex.Identifier] | None:
+        identifiers = []
+        if identifier_element := source_record.metadata.find("identifier", string=True):
+            identifiers.append(
+                timdex.Identifier(
+                    value=str(identifier_element.string),
+                    kind=identifier_element.get("identifierType") or "Not specified",
+                )
+            )
+        identifiers.extend(list(cls._get_alternateidentifiers(source_record)))
+        identifiers.extend(list(cls._get_relatedidentifiers(source_record)))
+        return identifiers or None
+
+    @classmethod
+    def _get_alternateidentifiers(
```

Comment:
Maybe another underscore, ` _get_alternate_identifiers()`?

Reply:
I was going for the actual `Datacite` element name but you're correct that `_get_alternate_identifiers` is easier to read. I'll update